src/main/java/com/squareup/kotlinpoet/CodeWriter.kt

```diff
@@ -317,7 +317,7 @@ internal class CodeWriter @JvmOverloads constructor(
     if (o is TypeSpec) {
       o.emit(this, null)
     } else if (o is AnnotationSpec) {
-      o.emit(this, true)
+      o.emit(this, inline = true, asParameter = true)
```

Comment:
Agreed, it's too generic here, however I didn't find a better option and was just relying on tests.